nomad/rpc.go

```diff
@@ -208,6 +208,11 @@ func (s *Server) handleMultiplex(ctx context.Context, conn net.Conn, rpcCtx *RPC
 	s.setupRpcServer(rpcServer, rpcCtx)
 
 	for {
+		// stop handling connections if context was cancelled
+		if ctx.Err() != nil {
+			return
```

Comment:
Nice catch. Can you add it to `handleMultiplexV2` as well

Reply:
Yep, will do!

Reply:
Resolved